$NetBSD: patch-poppler.defs,v 1.3 2016/03/29 21:19:56 khorben Exp $

Removed functions in poppler-0.16.

API change in poppler-0.15, see
http://cgit.freedesktop.org/poppler/poppler/commit/?id=b257428150e2c13dcc24fd8f75e4ee2c679ab414

Add fixes for poppler-0.39.0, see
https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=6fbcb03b918df48f33d24326640209de3d031074

--- poppler.defs.orig	2009-09-27 09:07:59.000000000 +0000
+++ poppler.defs
@@ -578,18 +578,6 @@
   )
 )
 
-(define-enum Orientation
-  (in-module "Poppler")
-  (c-name "PopplerOrientation")
-  (gtype-id "POPPLER_TYPE_ORIENTATION")
-  (values
-    '("portrait" "POPPLER_ORIENTATION_PORTRAIT")
-    '("landscape" "POPPLER_ORIENTATION_LANDSCAPE")
-    '("upsidedown" "POPPLER_ORIENTATION_UPSIDEDOWN")
-    '("seascape" "POPPLER_ORIENTATION_SEASCAPE")
-  )
-)
-
 (define-enum PageTransitionType
   (in-module "Poppler")
   (c-name "PopplerPageTransitionType")
@@ -1646,58 +1634,6 @@
   (return-type "GType")
 )
 
-(define-method render_to_pixbuf
-  (of-object "PopplerPage")
-  (c-name "poppler_page_render_to_pixbuf")
-  (return-type "none")
-  (parameters
-    '("int" "src_x")
-    '("int" "src_y")
-    '("int" "src_width")
-    '("int" "src_height")
-    '("double" "scale")
-    '("int" "rotation")
-    '("GdkPixbuf*" "pixbuf")
-  )
-)
-
-(define-method render_to_pixbuf_for_printing
-  (of-object "PopplerPage")
-  (c-name "poppler_page_render_to_pixbuf_for_printing")
-  (return-type "none")
-  (parameters
-    '("int" "src_x")
-    '("int" "src_y")
-    '("int" "src_width")
-    '("int" "src_height")
-    '("double" "scale")
-    '("int" "rotation")
-    '("GdkPixbuf*" "pixbuf")
-  )
-)
-
-(define-method get_thumbnail_pixbuf
-  (of-object "PopplerPage")
-  (c-name "poppler_page_get_thumbnail_pixbuf")
-  (return-type "GdkPixbuf*")
-)
-
-(define-method render_selection_to_pixbuf
-  (of-object "PopplerPage")
-  (c-name "poppler_page_render_selection_to_pixbuf")
-  (return-type "none")
-  (parameters
-    '("gdouble" "scale")
-    '("int" "rotation")
-    '("GdkPixbuf*" "pixbuf")
-    '("PopplerRectangle*" "selection")
-    '("PopplerRectangle*" "old_selection")
-    '("PopplerSelectionStyle" "style")
-    '("GdkColor*" "glyph_color")
-    '("GdkColor*" "background_color")
-  )
-)
-
 (define-method render
   (of-object "PopplerPage")
   (c-name "poppler_page_render")
@@ -1796,6 +1732,12 @@
   (of-object "PopplerPage")
   (c-name "poppler_page_get_text")
   (return-type "char*")
+)
+
+(define-method get_selected_text
+  (of-object "PopplerPage")
+  (c-name "poppler_page_get_selected_text")
+  (return-type "char*")
   (parameters
     '("PopplerSelectionStyle" "style")
     '("PopplerRectangle*" "rect")
